Source CertSpotter throwing issue

Running the tool as following throws error on certain domain. 
`subfinder -d domain.com`

```
panic: runtime error: index out of range [-1]

goroutine 36 [running]:
github.com/projectdiscovery/subfinder/pkg/subscraping/sources/certspotter.(*Source).Run.func1(0xc0000d84d0, 0xc00017c180, 0x7ffd9b838027, 0xb, 0xd35280)
	/home/vps/go/src/github.com/projectdiscovery/subfinder/pkg/subscraping/sources/certspotter/certspotter.go:52 +0xc6e
created by github.com/projectdiscovery/subfinder/pkg/subscraping/sources/certspotter.(*Source).Run
	/home/vps/go/src/github.com/projectdiscovery/subfinder/pkg/subscraping/sources/certspotter/certspotter.go:23 +0x89


```

I think the error is thrown when certspotter does not return any data. 

## How can we reproduce the issue?
1. To reproduce the issue try running the tool against non existing domain  (issue also occurs with existing domain though)
`subfinder -d a33weasdlsdf23wedsaaaaaaaaa.com`

## What are the running context details?
* Installation method (e.g. `pip`, `apt-get`, `git clone` or `zip`/`tar.gz`):  go get -u -v github.com/projectdiscovery/subfinder/cmd/subfinder
* Client OS  - Linux
* Program version (see banner):  v2
* Relevant console output (if any):

Yep, i was able to reproduce the issue.

```
panic: runtime error: index out of range

goroutine 66 [running]:
github.com/projectdiscovery/subfinder/pkg/subscraping/sources/certspotter.(*Source).Run.func1(0xc000138000, 0xc00019e000, 0xc000026fd0, 0xf, 0xd612a8)
        /root/go/src/github.com/projectdiscovery/subfinder/pkg/subscraping/sources/certspotter/certspotter.go:52 +0xce0
created by github.com/projectdiscovery/subfinder/pkg/subscraping/sources/certspotter.(*Source).Run
        /root/go/src/github.com/projectdiscovery/subfinder/pkg/subscraping/sources/certspotter/certspotter.go:23 +0x89
```

--- a/pkg/subscraping/sources/certspotter/certspotter.go
+++ b/pkg/subscraping/sources/certspotter/certspotter.go
@@ -49,6 +49,12 @@ func (s *Source) Run(ctx context.Context, domain string, session *subscraping.Se
 			}
 		}
 
+		// if the number of responses is zero, close the channel and return.
+		if len(response) == 0 {
+			close(results)
+			return
+		}
+
 		id := response[len(response)-1].ID
 		for {
 			reqURL := fmt.Sprintf("https://api.certspotter.com/v1/issuances?domain=%s&include_subdomains=true&expand=dns_names&after=%s", domain, id)

